src/dcrypto/v1/v1.go

```diff
@@ -91,19 +99,23 @@ func keys(pass, salt []byte, iterations int) (aesKey, hmacKey []byte, err error)
 	return aesKey, hmacKey, nil
 }
 
+// hashReadWriter hashes on write and on read finalizes the hash and returns it.
+// Writes after a Read will return an error.
 type hashReadWriter struct {
 	hash hash.Hash
 	done bool
 	sum  io.Reader
 }
 
```

Comment:
Above the comment `type hashReadWriter`, please add
```go
var _ io.ReadWriter = &hashReadWriter{}
```
This way we can always be sure that (*hashReadWriter) implements `io.Reader` and `io.Writer`.